jsonurl.d.ts

```diff
@@ -0,0 +1,64 @@
+interface JsonURLOptions {
+    // These are defined as protected in the comments
+    // setOrDefault(src, key, defValue);
+    // setOverride(src, key);
+    // setOrDefaultInt(src, key, defValue);
+}
+
+export interface JsonURLParseOptions extends JsonURLOptions {
+    //See https://github.com/jsonurl/jsonurl-js/blob/master/src/JsonURLParseOptions.js
+
+    //public properties
+    maxParseDepth?: number; //default 1 << 5 === 32
+    maxParseValues?: number; //default 1 << 12 === 4096
+    maxParseChars?: number; //default 1 << 15 === 32K
+    emptyValue?: any; //default is {}
+    nullValue?: any; //default is null
+
+    //These feel like they are intended to be protected
+    // impliedArray?: any[];
+    // impliedObject?: Record<string, any>;
+    // wwwFormUrlEncoded?: boolean;
+    // impliedStringLiterals?: boolean;
+    // allowEmptyUnquotedValues?: boolean;
+    // allowEmptyUnquotedKeys?: boolean;
+    // coerceNullToEmptyString?: boolean;
+    // noEmptyComposite?: boolean;
+    // AQF?: boolean;
+    // getMissingValue?: (key, pos) => any; //default throws an error
+}
+
+declare class JsonURL {
+    //Source comments suggest this is deprecated:
+    //* But in tests it is used.
+    //* It is necessary to parse()
+    constructor(prop?: JsonURLParseOptions);
+
+    //Source comments suggests static parse() is available,
```

Comment:
Yeah, the static method was added with a recent commit but I haven't cut a release yet.

I've chosen to deprecate (and eventually remove) the constructor and added the static parse() method so that the library interface mirrors JSON. That is, with static `parse()` and `stringify()` methods.